controllers/vspherecluster_reconciler.go

```diff
@@ -440,7 +440,7 @@ func (r clusterReconciler) reconcileVSphereClusterWhenAPIServerIsOnline(clusterC
 	go func() {
 		// Block until the target API server is online.
 		clusterCtx.Logger.Info("start polling API server for online check")
-		wait.PollUntilContextCancel(context.Background(), time.Second*1, true, func(context.Context) (bool, error) { return r.isAPIServerOnline(clusterCtx), nil }) //nolint:errcheck
+		wait.PollUntilContextCancel(context.Background(), time.Second*1, true, func(context.Context) (bool, error) { return r.isAPIServerOnline(ctx, clusterCtx), nil }) //nolint:errcheck
```

Comment:
We also have to use a background context for isAPIServerOnline and the other func below. Let's go with:

```go
		ctx := context.Background()
		// Block until the target API server is online.
		clusterCtx.Logger.Info("start polling API server for online check")
		wait.PollUntilContextCancel(ctx, time.Second*1, true, func(context.Context) (bool, error) { return r.isAPIServerOnline(ctx, clusterCtx), nil }) //nolint:errcheck
		clusterCtx.Logger.Info("stop polling API server for online check")
		clusterCtx.Logger.Info("triggering GenericEvent", "reason", "api-server-online")
		eventChannel := clusterCtx.GetGenericEventChannelFor(clusterCtx.VSphereCluster.GetObjectKind().GroupVersionKind())
		eventChannel <- event.GenericEvent{
			Object: clusterCtx.VSphereCluster,
		}

		// Once the control plane has been marked as initialized it is safe to
		// remove the key from the map that prevents multiple goroutines from
		// polling the API server to see if it is online.
		clusterCtx.Logger.Info("start polling for control plane initialized")
		wait.PollUntilContextCancel(ctx, time.Second*1, true, func(context.Context) (bool, error) { return r.isControlPlaneInitialized(ctx, clusterCtx), nil }) //nolint:errcheck
		clusterCtx.Logger.Info("stop polling for control plane initialized")
		apiServerTriggersMu.Lock()
		delete(apiServerTriggers, clusterCtx.Cluster.UID)
		apiServerTriggersMu.Unlock()
```

Reply:
Thanks for pointing it out!